### ISSUE TITLE
Avoid black blackground when zoom out

### ISSUE THREAD

**Issue by chrisail** [osmdroid/osmdroid#726]
## Issue Type
Question / Improvement

## Description and/or steps/code to reproduce the problem
Use the demo application and zoom out with pinch.
Background around scaled tiles is black.

### Version of osmdroid the issue relates to:
5.6.4

Is it possible to replace black background by tiles ?
If it's too complicated, is it possible to set a different background color ?

Is the solution close to rescaleCache function modification?
I made some tries, but with no success.

Thanks.
Chris
![zo](https://user-images.githubusercontent.com/18008107/31142699-5d267f40-a87b-11e7-97a8-7f2ca30efeff.png)

**Comment by spyhunter99** [comment]
mm you can set the back color by accessing the TilesOverlay class. The default is a gray grid. However in your case i believe the black is from the layout containing the mapview. Try setting the background color of the containing layout.

**Comment by chrisail** [comment]
Thank you spyhunter99, so sorry not to have seen this simple solution by myself.
As my TilesOverlay background color was correctly set, I was (thick headed) convinced that the problem was on ZoomOutTileLooper class and that black tiles were drawn by it.

As I would like to find a smarter way than just setting background color, I made a new and better code reading. So, now, I understand that ZoomOutTileLooper is just called at the end of pinch event (fingers released) just for displaying lower resolution bitmap before correct resolution is read from source.

I also understand that during a pinch (so without releasing fingers and for each pinch move) "performDragOrPinch()", method of "MultiTouchController" class, is called. This method then call "setPositionAndScale()", method of "MapView" class. 
This last method invalidate MapView that trigger "draw()" method of "TilesOverlay" class, that call "drawTiles()" method. It's end with a call to "loop()" method of "TileLooper". 

In this method, if I extend current viewPort like below:
```
TileSystem.PixelXYToTileXY(pViewPort.left - 2000, pViewPort.top - 2000, mUpperLeft);
//TileSystem.PixelXYToTileXY(pViewPort.left, pViewPort.top, mUpperLeft);
...
TileSystem.PixelXYToTileXY(pViewPort.right + 2000, pViewPort.bottom + 2000, mLowerRight);
//TileSystem.PixelXYToTileXY(pViewPort.right, pViewPort.bottom, mLowerRight);
```
I obtain the wished behavior, eg: **a larger map display during zoom control.** But it's just for understanding purpose and as this method is called for each type of draw, the solution is not fine and cause bad performances.

So, I would like during "performDragOrPinch()" enlarge display area and propagate it to "drawTiles()", but this data is lost during "draw()" call. I'm a little bit lost and I can't figure out how to do it.
Does I need to propagate myself this data or does I miss something in the code that allows me to do it natively ?
Thanks

**Comment by spyhunter99** [comment]
hmm i'm not really sure, maybe @monsieurtanuki knows.

**Comment by monsieurtanuki** [comment]
I think I fixed that in #329.
Anyway I cannot reproduce it anymore - though I remember I used to see black tiles on the borders when zooming out, especially with some map rotation.
@chrisail Could you check again with the latest version of osmdroid - or at least one that includes #329?

**Comment by chrisail** [comment]
Hi.
I've tested the **lastest commit** (74ae4be) of OpenStreetMapViewer and **it works fine**:
**Thank you** monsieurtanuki and also OSMDroid team for your great work.
FYI, there's a problem with the minimap position (red drawings), already present in 5.6.5.

Before during pinch zoom out with v 5.6.5, black around map (yellow arrows)
![unzoom](https://user-images.githubusercontent.com/18008107/31574432-1c3997da-b0d0-11e7-8221-187e70a8aca0.png)

Now during pinch zoom out with latest commit, no more black around map
![unzoom-fine](https://user-images.githubusercontent.com/18008107/31574405-74ff1c7e-b0cf-11e7-9605-798cd14518ee.png)
